openedx/openedx-platform
# [BB-2853] Add OAuth2-enabled endpoints for problem response report generation

Currently, the endpoints related to generation of problem response reports, only support session-based authentication, and cannot be easily used by MFE or other API clients. 

A [previous PR](https://github.com/edx/edx-platform/pull/19635) for this was reverted since it causes some issues. While we were not able to reproduce these issues, in the interest of disturbing as little of the existing code as possible, this PR retains the existing endpoints, but adds alternative endpoints that expose the same views but through a DRF setup that supports OAuth2-based authentication.


**Discussions**: 
- https://github.com/edx/edx-platform/pull/19635 
- https://github.com/edx/edx-platform/pull/24873
- This PR is following the pattern detailed in this ADR PR:https://github.com/edx/edx-platform/pull/27129

**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: "None"

**Testing instructions**:

1. ``paer test_system -t lms/djangoapps/instructor/tests/test_api.py``
2. Ensure that the existing instructor dashboard report generation works by generating a problem response report, and seeing that it runs properly and creates a new report.
3. Create an OAuth2 token. You can either create a new OAuth2 client, and use the OAuth2 api to generate a token. Or just add one manually here; /admin/oauth2_provider/accesstoken/add/ 
   Make sure to use a user that has permissions to generate a report, such as a staff user/ 
4. Use postman/httpie/curl etc to call the `generate_problem_responses` endpoint for a course with a corresponding `problem_location`. This will return a task id. 
5. Call the `list_instructor_task` endpoint for the same course. Usually, it is hard to observe the task actually in progress in the devstack, without too much data. The endpoint should work with OAuth though, which is the main thing to test. 
6. Call the `list_report_downloads` endpoint. It should include the newly-generated report. 



**Reviewers**

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true
```
